### PR TITLE
add readme for e2e tests

### PR DESCRIPTION
**Pull request by brycahta**
Issue #, if available: N/A

Description of changes: 
* Adding readme to demystify e2e tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+**The Verification**
+* After the *system-reboot* event, the test will verify that the node being rebooted has been cordoned, tainted properly (no new work to be scheduled), and any running pods are evicted (existing work is removed).
+* Once the nodes and pods are in expected state, the *canceled* event is sent and the test verifies the node is no longer cordoned by checking its status AND scheduling another pod (new work) successfully.
+
+**The Walkthrough**
+* NTH and an empty test pod, regular-pod-test, are deployed to the cluster
+  *  regular-pod-test is used to represent work scheduled onto the nodes
+* Regular-pod-test should be scheduled and started on the worker node
+* [EC2-Metadata-Mock](https://github.com/aws/amazon-ec2-metadata-mock) (aka AEMM; used to mock IMDS) is installed on the cluster
+* When AEMM is starts running, a new maintenance event for *system-reboot* will be sent and consumed by NTH.
```
Comment (bwagner5) [inline test/README.md:50]: "When AEMM starts running"

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+* NTH consumes the event and forwards the *system-reboot* event to its [interruption channel](https://github.com/aws/aws-node-termination-handler/blob/18ce8fdd87172c5e774b3693b29ce62c49e93272/cmd/node-termination-handler.go#L152) triggering the cordoning of the node and draining of any pods
+* Once nodes have been cordoned and pods drained, AEMM is restarted to send a *canceled* event instead of system-reboot
+* NTH forwards the *canceled* event to its [cancel channel](https://github.com/aws/aws-node-termination-handler/blob/18ce8fdd87172c5e774b3693b29ce62c49e93272/cmd/node-termination-handler.go#L160) which proceeds with uncordoning and removing any taints
+* If the node is uncordoned and taints were removed successfully, then new work (pods) should be schedulable to the node
+* Therefore, the final check is ensuring regular-pod-check is rescheduled and deployed to the node
```
Comment (bwagner5) [inline test/README.md:55]: regular-pod-test

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+#### TL;DR
+Instead of expected/actual *values*, the tests evaluate expected/actual **states**. NTH's functionality is evaluated by triggering various interruption events on the cluster (spot interrupts, maintenance events) and ensuring the nodes and pods in the cluster are in their proper state (cordoned, tainted, evicted, etc.). The tests fail if the nodes/pods don't transition into expected states suggesting an issue in NTH logic or cluster setup.
+
+
+#### Summary
+This doc details how the end-to-end (e2e) tests work for aws-node-termination-handler (NTH) at a high-level. These tests are no different from normal integration tests in that they capture the functionality of NTH. However, the *assert-actual-equals-expected* pattern is not as explicit as in other e2e tests which can cause some confusions. We hope to bring clarity with the content below.
+
+
+#### Starting Tests
```
Comment (bwagner5) [inline test/README.md:12]: can you add an example of running the drivers directly and using the preserve flag?  I think most of the people reading this README will be devs working on a PR and the preserve example would be extremely valuable for them.

Comment (brycahta) [reply test/README.md:12]: added **Using Test Drivers** `Starting Tests` section

**Pull request merged**